File: 0228-summary-ranges/0228-summary-ranges.cpp
class Solution {
public:
    vector<string> summaryRanges(vector<int>& nums) {
        vector<string> answer; 
        int n = nums.size(); 
        for(int i = 0; i < n; i++){
            int curr = nums[i]; 
            int next = nums[i]; 
            while(i < n -1 && nums[i+1] == (nums[i] + 1)){
                next = nums[i+1];
                i++;
            }

            if(curr != next){
                answer.push_back(to_string(curr) + "->" + to_string(next)); 
            } else{
                answer.push_back(to_string(curr));  
            }
        }
        return answer; 
    }
};

File: 0228-summary-ranges/0228-summary-ranges.java
class Solution {
    public List<String> summaryRanges(int[] nums) {
        List<String> answer = new ArrayList<>();
        int start = 0, end = 0; 
        StringBuilder sb = new StringBuilder(); 
        while(end < nums.length){
            int curr = nums[end]; 
            int last = nums[end]; 
            while(end + 1 < nums.length && nums[end+1] - nums[end] == 1){
                end++; 
                last = nums[end]; 
            }

            if(curr != last){
                sb.append(Integer.toString(curr)).append("->").append(Integer.toString(last));
            } else{
                sb.append(Integer.toString(curr)); 
            }

            answer.add(sb.toString()); 
            sb = new StringBuilder(); 
            end++; 
        }
        return answer; 
    }
}